src/ExportMap.js

```diff
@@ -781,12 +782,29 @@ export function recursivePatternCapture(pattern, callback) {
   }
 }
 
+let parserOptionsHash = '';
+let prevParserOptions = null;
+let settingsHash = '';
+let prevSettings = null;
 /**
  * don't hold full context object in memory, just grab what we need.
+ * also calculate a cacheKey, where parts of the cacheKey hash are memoized
  */
 function childContext(path, context) {
   const { settings, parserOptions, parserPath } = context;
+
+  if (!deepEqual(settings, prevSettings)) {
```

Comment:
why not `JSON.stringify(settings) !== JSON.stringify(prevSettings)`?

Reply:
@ljharb Pushed a version without `fast-deep-equal` and utilizing `JSON.stringify`.

It unnecessarily creates ~1M objects and therefore seems to take 1.9s instead of 0.6s. But we still improve local execution by ~10s.

Reply:
(updated the numbers in the issue description)

Reply:
I'm surprised JSON.stringify would create any objects at all, but improving execution sounds good.